parachain/src/wasm_executor/validation_host.rs

```diff
@@ -169,22 +203,30 @@ pub fn run_worker(mem_id: &str) -> Result<(), String> {
 		{
 			debug!(target: LOG_TARGET, "{} Processing candidate", process::id());
 			// we have candidate data
-			let mut slice = memory.wlock_as_slice(0)
-				.map_err(|e| format!("Error locking shared memory: {:?}", e))?;
-
 			let result = {
-				let data: &mut[u8] = &mut **slice;
-				let (header_buf, rest) = data.split_at_mut(1024);
+				// Locking as read only to allow access to code memory from
+				// host validation extension.
+				// Note that this only work if two process are communicating only.
+				// Would be racy otherwhise.
```

Comment:
I am unsure of this assertion: I did switch the lock to read to be able to acces the wasm slice directly, but this would be racy if the shared memory is consumed by multiple client.
CC\ @arkpar

Reply:
I don't understand why you are doing all this, when we read the code anyway.

Reply:
The code is passed as a slice and the validation host function uses an extension which require 'static lifetime, so I cannot use a slice in it.

Reply:
I think it is better to clone than doing the magic here.

Reply:
I can also try  to access it with 'let validation_code = runtime_api_request( ctx, descriptor.relay_parent, RuntimeApiRequest::ValidationCode( descriptor.para_id, assumption, code_tx, ), code_rx, ).await?;', not sure it if it is accessible.

Reply:
Multiple parallel validation are supported, but each process is run in a separate instance of `ValidationHost`. So aliasing code memory slice should be safe.